### PR TITLE
IconButton: fix mousearea and remove animation

### PR DESCRIPTION
**Pull request by selsta**
The animation was buggy if the mouse got released mouse outside of the mouse-area or outside of the whole program. Maybe someone has an idea how this can get fixed instead of removing it?

### CODE REVIEW

**Comment by mmbyday** [comment]
This will keep the animation intact and fix the release bug.
```
diff --git a/components/IconButton.qml b/components/IconButton.qml
index 801c01e..de82660 100644
--- a/components/IconButton.qml
+++ b/components/IconButton.qml
@@ -35,7 +35,6 @@ Item {

     signal clicked(var mouse)

-
     id: button
     width: parent.height
     height: parent.height
@@ -47,13 +46,13 @@ Item {
         id: buttonImage
         source: ""
         x : (parent.width - width) / 2
-        y : (parent.height - height)  /2
+        y : (parent.height - height) / 2
         z: 100
     }

     MouseArea {
         id: buttonArea
-        anchors.fill: parent
+        anchors.fill: buttonImage
         hoverEnabled: true
         cursorShape: Qt.PointingHandCursor

@@ -61,14 +60,14 @@ Item {
             buttonImage.x = buttonImage.x + 2
             buttonImage.y = buttonImage.y + 2
         }
+
         onReleased: {
-            buttonImage.x = buttonImage.x - 2
-            buttonImage.y = buttonImage.y - 2
+            buttonImage.x = (parent.width - width) / 2
+            buttonImage.y = (parent.height - height) / 2
         }

         onClicked: {
             parent.clicked(mouse)
         }
     }
-
 }
```

**Comment by selsta** [comment]
thanks, updated.

**Comment by selsta** [comment]
@mmbyday Somehow this is still buggy for me :/

File: components/IconButton.qml [hunk 1]
```diff
@@ -47,28 +46,29 @@ Item {
...
-        y : (parent.height - height)  /2
+        y : (parent.height - height) / 2
         z: 100
     }
 
     MouseArea {
         id: buttonArea
-        anchors.fill: parent
+        anchors.fill: buttonImage
+        anchors.margins: -5
```
Comment (mmbyday) [inline components/IconButton.qml]:
```suggestion
```
This was the culprit :b

Comment (selsta) [reply components/IconButton.qml]: That’s to make the clickable area a little big bigger than the icon. :)

**Pull request merged**